eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -0,0 +1,26 @@
+parameters:
+  Repos: []
+  DirectoryToSync: eng/common
+  CommitMessage: commit-message-not-set
+
+steps:
+
+- ${{ each repo in parameters.Repos }}:
+  - pwsh: |
+      git clone --branch master https://github.com/azure/${{ repo }}
```

Comment:
What about a shallow clone for speed?

Reply:
I tried to do a shallow clone but that doesn't work in our case because we need the full source history to be able to make commits and push them.